plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs

```diff
@@ -333,6 +340,28 @@ instance uni ~ DefaultUni => ToBuiltinMeaning uni DefaultFun where
         makeBuiltinMeaning
             (\() a -> a)
             mempty
+    toBuiltinMeaning MkPairData =
+        makeBuiltinMeaning
+            ((,) :: Data -> Data -> (Data, Data))
+            mempty
+    toBuiltinMeaning MkNilData =
+        -- Nullary builtins don't work, so we need a unit argument
+        makeBuiltinMeaning
+            ((\() -> []) :: () -> [Data])
+            mempty
+    toBuiltinMeaning MkConsData =
+        makeBuiltinMeaning
+            ((:) :: Data -> [Data] -> [Data])
+            mempty
+    toBuiltinMeaning MkNilPairData =
+        -- Nullary builtins don't work, so we need a unit argument
+        makeBuiltinMeaning
+            ((\() -> []) :: () -> [(Data,Data)])
```

Comment:
That's quite annoying to parse with eyes. I'd write it as

```haskell
        makeBuiltinMeaning
            @(() -> [(Data,Data)])
            (\() -> [])
```

And others too.